File: gb-angular-frontend/src/app/interfaces/cart-item.ts
import {Package} from './package';
import {PackageItem} from './package-item';

export interface CartItem {
  cart_id?: number;
  buyer_id: number;
  product_id: number;
  product_title: string;
  package_id?: number
  package_name?: string;
  seller_id?: number;
  seller_username?: string;
  quantity: number;
  unit_price: number;
  sub_Total: number;
  sell_type: string;
  item_image_url: string;
  package_grp_id?: number;
  package_grp_name?: string;
  grpbuy_end?: string;
  package_item_list?: PackageItem[];
}

File: gb-spring-backend/src/main/java/com/egroupbuy/model/CartItem.java
package com.egroupbuy.model;

import lombok.*;

import java.sql.Date;
import java.sql.Timestamp;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class CartItem {
    int cart_id;
    int buyer_id;
    int product_id;
    String product_title;
    int package_id;
    String package_name;
    int seller_id;
    String seller_username;
    int quantity;
    double unit_price;
    double sub_Total;
    String sell_type;
    String item_image_url;
    int package_grp_id;
    String package_grp_name;
    Timestamp grpbuy_end;
    List<PackageItem> package_item_list;
}
